docs/advanced_usage.rst

```diff
@@ -602,6 +602,20 @@ traces should be sent off. If an error occurs, a message will be displayed in
 the console, and changes can be made as needed.
 
 
+uWSGI
+-----
+
+The default configuration of uWSGI applications does not include the
+``--enable-threads`` setting which must be set to ``true`` for the
+tracing library to run.  This is noted in their best practices doc_.
+
+  .. _doc: https://uwsgi-docs.readthedocs.io/en/latest/ThingsToKnow.html
+
+Example run command:
+
+``ddtrace-run uwsgi --http :9090 --wsgi-file your_app.py --enable-threads``
```

Comment:
should we show an example with and without `ddtrace-run`?

not actually sure it matters here, but /shrug ?

Reply:
@majorgreys and I were discussing that and weren't sure it made sense to give that much instruction here since using patching is not our recommended approach and users would ultimately need to read the advanced usage docs to get started.

https://docs.datadoghq.com/tracing/manual_instrumentation/python?tab=decorator
http://pypi.datadoghq.com/trace/docs/advanced_usage.html#patch-all

I don't immediately see a key difference in how someone would configure uWSGI due to a decision to use the more manual approach.  One would start their uwsgi app as normal without prepending `dd-trace run` and then configure in the necessary `from ddtrace import tracer` use the `patch()` function and then use wrap() etc if they wished to apply any manual instrumentation.  Does that sound correct?

Reply:
I was considering quoting this section from the uWSGI best practices doc hyperlinked above since it is rather long and verbose for a reader to check at a glance.

> By default the Python plugin does not initialize the GIL. This means your app-generated threads will not run. If you need threads, remember to enable them with enable-threads. Running uWSGI in multithreading mode (with the threads options) will automatically enable threading support. This “strange” default behaviour is for performance reasons, no shame in that.

Reply:
👍